src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerServerCredentialsBinding.java

```diff
@@ -0,0 +1,59 @@
+package org.jenkinsci.plugins.docker.commons.credentials;
+
+import java.io.IOException;
+
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.credentialsbinding.BindingDescriptor;
+import org.jenkinsci.plugins.credentialsbinding.impl.AbstractOnDiskBinding;
+import org.jenkinsci.plugins.docker.commons.Messages;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import hudson.Extension;
+import hudson.FilePath;
+
+public class DockerServerCredentialsBinding extends AbstractOnDiskBinding<DockerServerCredentials> {
+
+    @DataBoundConstructor
+    public DockerServerCredentialsBinding(String variable, String credentialsId) {
+        super(variable, credentialsId);
```

Comment:
I like the idea, but, well, I will pass. Lazy me...

I've pushed fixes for your other review comments.